src/server/pps/server/poller.go

```diff
@@ -284,26 +284,19 @@ func (m *ppsMaster) watchPipelines(ctx context.Context) {
 			if err != nil {
 				return errors.Wrap(err, "bad watch event key")
 			}
-			var e *pipelineEvent
 			switch event.Type {
-			case watch.EventPut:
-				e = &pipelineEvent{
-					eventType: writeEv,
+			case watch.EventPut, watch.EventDelete:
+				e := &pipelineEvent{
 					pipeline:  pipelineName,
 					timestamp: time.Unix(event.Rev, 0),
 				}
-			case watch.EventDelete:
-				e = &pipelineEvent{
-					eventType: deleteEv,
-					pipeline:  pipelineName,
-					timestamp: time.Unix(event.Rev, 0),
+				select {
+				case m.eventCh <- e:
+				case <-m.masterCtx.Done():
+					return errors.Wrap(err, "pipeline event arrived while master is restarting")
 				}
-			}
-			select {
-			case m.eventCh <- e:
-				continue
-			case <-m.masterCtx.Done():
-				return errors.Wrap(err, "pipeline event arrived while master is restarting")
+			case watch.EventError:
```

Comment:
we weren't handling this case before. On errors we would silently send a nil event to the master